src/salte-auth.js

```diff
@@ -1,1226 +1,324 @@
-   */
-  getCachedUser() {
-    if (this._user) {
-      return this._user;
+    let authorizeGateway = `${this.$config.gateway}/authorize`;
+    if (this.provider.authorizeUrl) {
+      authorizeGateway = this.provider.authorizeUrl.call(this, this.$config);
     }
 
-    const idtoken = this._getItem(this.CONSTANTS.STORAGE.IDTOKEN);
-    this._user = this._createUser(idtoken);
-    return this._user;
-  }
-
-  registerCallback(expectedState, resource, callback) {
-    this._activeRenewals[resource] = expectedState;
-    if (!this.callBacksMappedToRenewStates[expectedState]) {
-      this.callBacksMappedToRenewStates[expectedState] = [];
-    }
-    this.callBacksMappedToRenewStates[expectedState].push(callback);
-    if (!this.callBackMappedToRenewStates[expectedState]) {
-      this.callBackMappedToRenewStates[expectedState] = (message, token) => {
-        for (let i = 0; i < this.callBacksMappedToRenewStates[expectedState].length; ++i) {
-          try {
-            this.callBacksMappedToRenewStates[expectedState][i](message, token);
-          } catch (error) {
-            this.warn(error);
-          }
-        }
-        this._activeRenewals[resource] = null;
-        this.callBacksMappedToRenewStates[expectedState] = null;
-        this.callBackMappedToRenewStates[expectedState] = null;
-      };
-    }
+    return this.utilities.createUrl(authorizeGateway, assign({
+      'state': this.profile.localState,
+      'nonce': this.profile.nonce,
+      'response_type': 'token',
+      'redirect_uri': this.$config.redirectUrl,
+      'client_id': this.$config.clientId,
+      'scope': this.$config.scope,
+      'prompt': 'none'
+    }, this.$config.queryParams));
   }
 
   /**
-   * Acquires access token with hidden iframe
-   * @param {string} resource  ResourceUri identifying the target resource
-   * @param {string} callback  ResourceUri identifying the target resource
+   * The authentication url to retrieve the id token
+   * @type {String}
    */
-  _renewToken(resource, callback) {
-    // use iframe to try refresh token
-    // use given resource to create new authz url
-    this.info('renewToken is called for resource:' + resource);
-    const frameHandle = this._addAuthFrame('authRenewFrame' + resource);
-    const expectedState = uuid.v4() + '|' + resource;
-    this.config.state = expectedState;
-    // renew happens in iframe, so it keeps javascript context
-    this._renewStates.push(expectedState);
-
-    this.verbose('Renew token Expected state: ' + expectedState);
-    let urlNavigate = this._getNavigateUrl('token', resource) + '&prompt=none';
-    urlNavigate = this._addHintParameters(urlNavigate);
-
-    this.registerCallback(expectedState, resource, callback);
-    this.verbose('Navigate to:' + urlNavigate);
-    this._saveItem(this.CONSTANTS.STORAGE.LOGIN_REQUEST, '');
-    frameHandle.src = 'about:blank';
-    this._loadFrameTimeout(urlNavigate, 'authRenewFrame' + resource, resource);
-  }
+  get authorizeUrl() {
```

Comment:
I think this should be renamed to idTokenUrl to be consistent with the getter for accessTokenUrl.  It is also confusing because both salte-auth and the provider can currently have something called an authorizeUrl.